tests/test_docker_cat.py

```diff
@@ -0,0 +1,539 @@
+        """
+        sensors = (
+            "INFO: Sensor CheckstyleSensor [checkstyle]",
+            "INFO: Sensor FindBugs Sensor [findbugs]",
+            "INFO: Sensor PmdSensor [pmd]",
+            "INFO: Sensor CoberturaSensor [cobertura]"
+        )
+        self.language("Java", "java", "java", sensors, "java-dummy-project", 3, "CNES_JAVA_A", 6)
+
+    def test_language_python(self):
+        """As a user of this image, I want to analyze a Python project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        self.language("Python", "py", "python", (), "python-dummy-project", 2, "CNES_PYTHON_A", 3)
+
+    def test_language_shell(self):
+        """As a user of this image, I want to analyze a shell project
+        so that I can see its level of quality on the SonarQube server.
+        """
+        sensors = (
+            "INFO: Sensor Sonar i-Code [icode]",
+        )
+        self.language("Shell", "shell", "shell", sensors, "shell-dummy-project", 58)
+
+    # Test analysis tools
+    def test_tool_cppcheck(self):
+        """As a user of this image, I want to run cppcheck from within a container
+        so that it produces a report.
+        """
+        ref = "tests/c_cpp/reference-cppcheck-results.xml"
+        output = "tests/c_cpp/tmp-cppcheck-results.xml"
+        cmd = f"cppcheck --xml-version=2 tests/c_cpp/cppcheck/main.c --output-file={output}"
+        self.analysis_tool("cppcheck", cmd, ref, output, False)
+
+    def test_tool_frama_c(self):
+        """As a user of this image, I want to run Frama-C from within a container
+        so that it produces a report.
+        """
+        ref = "tests/c_cpp/reference-framac-results.txt"
+        output = "tests/c_cpp/tmp-framac-results.txt"
+        report = "tests/c_cpp/frama-c.csv"
+        cmd = f"frama-c tests/c_cpp/framac/CruiseControl.c tests/c_cpp/framac/CruiseControl_const.c -rte -metrics -report-csv {report}"
+        self.analysis_tool("Frama-C", cmd, ref, output)
+
+    def test_tool_pylint(self):
+        """As a user of this image, I want to run pylint from within a container
+        so that it produces a report.
+        """
+        cmd = "pylint --exit-zero -f json --rcfile=/opt/python/pylintrc_RNC_sonar_2017_A_B tests/python/src/simplecaesar.py"
+        self.analysis_tool("pylint", cmd, "tests/python/reference-pylint-results.json", "tests/python/tmp-pylint-results.json")
+
+    def test_tool_rats(self):
+        """As a user of this image, I want to run RATS from within a container
+        so that it produces a report.
+        """
+        ref = "tests/c_cpp/reference-rats-results.xml"
+        output = "tests/c_cpp/tmp-rats-results.xml"
+        cmd = "rats --quiet --nofooter --xml -w 3 tests/c_cpp/rats"
+        self.analysis_tool("RATS", cmd, ref, output)
+
+    def test_tool_shellcheck(self):
+        """As a user of this image, I want to run shellcheck from within a container
```

Comment:
Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/docker-cat/pullRequest?prid=6253839)